dev/source-release.sh

```diff
@@ -66,7 +66,11 @@ tarball=$tag.tar.gz
 
 # be conservative and use the release hash, even though git produces the same
 # archive (identical hashes) using the scm tag
-git archive $release_hash --prefix $tag/ -o $tarball .baseline api arrow bundled-guava common core data dev flink gradle gradlew hive mr orc parquet pig spark spark2 spark-runtime spark3 spark3-runtime LICENSE NOTICE README.md build.gradle baseline.gradle deploy.gradle tasks.gradle jmh.gradle gradle.properties settings.gradle versions.lock versions.props version.txt
+adds=" .baseline"  # prefixed with a blank space for each file name including the first one
+excludes="build|examples|jitpack.yml|python|site"
+archives=$(ls | grep -vE ${excludes})${adds}
+echo git archive list: ${archives}
```

Comment:
So perhaps I'm running this wrong, but my `$archives` comes out formatted kind of funny based on my expectations. Everything is separated by a newline character except for ` .baseline` which is separated by a space from the last entry. I believe you're correct that `git-archive` excpects the output of archive files / directories to be separated by spaces. Is it possible that we need to munge the output of `$archives` to be space delimited instead of the mixture of newline delimited and space delimited that I got? It's also possible that I ran the script wrong.

Below is my $archives list after running this portion of the script using `apache-iceberg-0.9.0` release (I couldn't find any releases tagged with as an `rc` candidate though I didn't spend a ton of time looking.

```bash
$ echo ${archives}
LICENSE
NOTICE
README.md
api
arrow
baseline.gradle
bundled-guava
common
core
data
deploy.gradle
dev
flink
gradle
gradle.properties
gradlew
hive
jmh.gradle
mr
orc
parquet
pig
settings.gradle
spark
spark-runtime
spark2
spark3
spark3-runtime
tasks.gradle
versions.props .baseline
```

Reply:
If I add `archives=$(echo $archives | tr '\n' ' ')`, then my `$archives` output is space delimited as I believe is the expected behavior for input to `git archive`.

```bash
$ echo ${archives}
LICENSE
NOTICE
README.md
api
arrow
baseline.gradle
bundled-guava
common
core
data
deploy.gradle
dev
flink
gradle
gradle.properties
gradlew
hive
jmh.gradle
mr
orc
parquet
pig
settings.gradle
spark
spark-runtime
spark2
spark3
spark3-runtime
tasks.gradle
versions.props .baseline

$ archives=$(echo $archives | tr '\n' ' ')
$ echo $archives
LICENSE NOTICE README.md api arrow baseline.gradle bundled-guava common core data deploy.gradle dev flink gradle gradle.properties gradlew hive jmh.gradle mr orc parquet pig settings.gradle spark spark-runtime spark2 spark3 spark3-runtime tasks.gradle versions.props .baseline 
```

Reply:
@kbendick Really sorry for my late response~~ Thanks for pointing that out. I never noticed that.
But it seems a happy accident that it has the expected output...
I made the following experiments just now:
1. `echo ${archives}` directly (not in a shell script) -> the output is new line delimited.
2. `echo "${archives}"` directly (not in a shell script) -> the output is new line delimited. 
3. `echo ${archives}` in a shell sript -> the output is **space** delimited   <-- my lucky
4. `echo "${archives}"` in a shell sript -> the output is new line delimited
Yes, it is more safe to normalize the input to be all space delimited

Reply:
Ahh I see. It's a good thing we checked on another shell then. It's probably safer to normalize to ensure the script is more portable.

Reply:
Ahh I see. It's a good thing we checked on another shell then. It's probably safer to normalize to ensure the script is more portable.

Reply:
Marking this as closed as this has been fixed.